### PR TITLE
Add InjectSpanContext ClientOption.

### LINKED ISSUE #43: Make injecting tracing context optional
When connecting to external services sending the tracing context in HTTP headers provides no value and should be able to be suppressed. The client side tracing is still however useful, so it is not desirable to simply not trace the request at all.

### PR DESCRIPTION
**Pull request by nvx**
Fixes #43.

### CODE REVIEW

File: nethttp/client.go [hunk 1]
```diff
@@ -64,6 +65,14 @@ func ClientTrace(enabled bool) ClientOption {
 	}
 }
 
+// InjectSpanContext returns a ClientOption that turns on or off
+// injection of the Span context in the request HTTP headers.
```
Comment (yurishkuro) [inline nethttp/client.go]: add: If this option is not used, the default behavior is to inject span context.

**Comment by nvx** [comment]
Documentation updated. As for tests, I looked to see if there was an existing test that verifies the existing behaviour that headers are added correctly but couldn't spot one. Ideally the test for this PR would be to check the opposite happens when expected and that the existing behaviour is unchanged otherwise.

Due the absence of a test to check that the headers are correctly added normally, I'm not sure what the kosher way to do that would be. I can simply check for the presence or absence of some fixed header fields but noting that the library doesn't add them directly (it just calls Inject) I'm not sure if this is really the right thing to be testing. Alternatively could mock the Inject() call, but it's a few levels deep and also smells a bit weird. Thoughts?

**Comment by yurishkuro** [comment]
Traditionally the MockTracer from OpenTracing is used to verify instrumentation. Rather that looking for headers, I usually start client and server and verify that the span created in the server has the same trace id as in the client. In case of "do not inject" flag, trace IDs should be different.

**Comment by nvx** [comment]
How's that look? Tests both explicit true and false, as well as the default when not specified (implicit enable injection)

File: nethttp/client.go [hunk 2]
```diff
@@ -58,12 +59,22 @@ func ComponentName(componentName string) ClientOption {
 
 // ClientTrace returns a ClientOption that turns on or off
 // extra instrumentation via httptrace.WithClientTrace.
+// If this option is not used, the default behaviour is to
+// inject the span context.
```
Comment (yurishkuro) [inline nethttp/client.go]: did you mean to add this comment to InjectSpanContext ?

Comment (nvx) [reply nethttp/client.go]: Good catch!

**Pull request merged**